ext/spl/spl_iterators.c

```diff
@@ -3614,12 +3626,60 @@ PHP_FUNCTION(iterator_to_array)
 		RETURN_FALSE;
 	}
 
-	array_init(return_value);
+	if (spl_iterator_to_array_do_convert(obj, use_keys, return_value) == 0) {
+		RETURN_NULL();
+	}
+} /* }}} */
+
+/* {{{ proto array iterable_to_array(iterable it [, bool use_keys = true])
+   Convert an iterable into an array */
+PHP_FUNCTION(iterable_to_array)
+{
+	zval  *iter, *entry;
+	zend_bool use_keys = 1;
+	zend_long arrlen;
+
+	if (zend_parse_parameters(ZEND_NUM_ARGS(), "z|b", &iter, &use_keys) == FAILURE) {
+		RETURN_FALSE;
+	}
 
-	if (spl_iterator_apply(obj, use_keys ? spl_iterator_to_array_apply : spl_iterator_to_values_apply, (void*)return_value) != SUCCESS) {
-		zval_ptr_dtor(return_value);
+	if (zend_is_iterable(iter) == 0) {
+		zend_type_error("Argument 1 passed to iterable_to_array() must be iterable, %s given",
```

Comment:
Don't we have this treatment in ZPP?

Reply:
There doesn't seem to be a type specifier for `iterable` currently: http://php.net/manual/en/internals2.funcs.php
Also checked source directly and didn't see anything obvious. :/

Reply:
Maybe we should add it :)